src/braket/emulators/emulator_interface.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from abc import ABC
+from typing import Iterable, Union
+
+from braket.emulators.emulator_passes import EmulatorPass, ProgramType
+from braket.emulators.emulator_passes.criteria import EmulatorCriterion
+
+
+class EmulatorInterface(ABC):
```

Comment:
Is this really an `ABC` (let alone an interface) if there are no abstract methods? Maybe just call it `BaseEmulator`?

Reply:
This makes sense, `EmulatorInterface` is definitely not an appropriate name for this class - changed to `BaseEmulator`.